lib/zlib.js

```diff
@@ -114,7 +119,7 @@ function zlibBuffer(engine, buffer, callback) {
     var err = null;
 
     if (nread >= kMaxLength) {
-      err = new RangeError(kRangeErrorMessage);
+      err = new errors.RangeError('ERR_INVALID_RANGE', kRangeErrorMessage);
```

Comment:
This is somewhat specific to Buffers, though… maybe a name like `ERR_EXCEEDS_MAX_BUFFER_LEN(GTH)` would be better?